mitodl/mit-open
# Don't display carousel tabs if there's no data to display

### What are the relevant tickets?

Closes mitodl/hq#4528

### Description (What does it do?)

Adds a new tab button component for the featured courses that can load the data that will be displayed in the tab, and then will disable itself if there's no content for the tab. 

This also fixes a small issue with the regex validator for channels - they get created with names that have `-` in them, but the validator didn't include that so you can't edit channels that were created via pipeline. 

### How can this be tested?

Automated tests should pass.

Populate your featured lists with 0 items: `./manage.py populate_featured_lists 0`
You should see no tabs in Featured Courses.

Then, do the same thing but add some stuff: `./manage.py populate_featured_lists 10`
You should see the tabs appear now.

### Additional Context

The tabs buttons do the loading now - so they appear as the data is retrieved. Locally, this means they pop in sorta slowly as for whatever reason these API hits take a while. Not sure if it would be better to have them load the tab, then remove themselves if they load 0 resources (it seems kinda disruptive either way). 

This may highlight efficiency issues with the featured resources API - as noted, it's pretty slow for me but I'm not sure if that's the API being slow or if it's just Docker networking on macOS (which is pretty poky anyway).
